src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,137 @@
+import duke.Parser;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+import duke.DukeException;
+
+import java.util.Scanner;
+
+/**
+ * Responsible for interpreting the input and interacting with the User.
+ */
+public class Duke {
+    private final Ui ui;
+    private TaskList taskList;
```

Comment:
The textbook says: "Distinguish clearly between single-valued and multivalued variables."
Here I assume the taskList should be many tasks inside the list?
So perhaps it should be taskLists instead of taskList

Reply:
Yup. You are right. I have made amendment for all the suggestions that you made.

Thanks.